api/src/main/java/net/kyori/adventure/util/HSVLikeImpl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2021 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.util;
+
+import net.kyori.examination.string.StringExaminer;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.Objects;
+
+final class HSVLikeImpl implements HSVLike {
+  private final float h;
+  private final float s;
+  private final float v;
+
+  HSVLikeImpl(final float h, final float s, final float v) {
+    this.h = h;
+    this.s = s;
+    this.v = v;
+  }
+
+  @Override
+  public float h() {
+    return this.h;
+  }
+
+  @Override
+  public float s() {
+    return this.s;
+  }
+
+  @Override
+  public float v() {
+    return this.v;
+  }
+
+  @Override
+  public boolean equals(final @Nullable Object other) {
+    if(this == other) return true;
+    if(!(other instanceof HSVLikeImpl)) return false;
+    final HSVLikeImpl that = (HSVLikeImpl) other;
+    return Float.compare(that.h, this.h) == 0 && Float.compare(that.s, this.s) == 0 && Float.compare(that.v, this.v) == 0;
```

Comment:
Should be able to use [`ShadyPines.equals`](https://github.com/KyoriPowered/adventure/blob/master/api/src/main/java/net/kyori/adventure/util/ShadyPines.java#L69-L79) here?

Reply:
Didn't know of that method until now, but looks like yes